### DISCUSSION TITLE
Is there a way to integrate supabase into Roblox LUA?

### DISCUSSION THREAD

**Discussion by LawMixer** [supabase/supabase#2917] [Questions] [answered]
I am currently trying to find another good database, that doesn't limit me just one database and actually has good staff, etc to move all of my products too! 

I am wondering if you guys support; 

- Roblox LUA (LUAU)
- Discord Bot Maker (DBM)

**Comment by silentworks** [comment, marked as answer]
You can use Supabase with LUA if they have an HTTP client library, this would however require you to create your own calls to the RESTful endpoints and handle everything yourself. Maybe someone from the community will create an LUA library one day (maybe this could be you).

The Discord Bot Maker thing looks like a bespoke piece of software, it would be down to the creator of the software to implement this.

**Reply by LawMixer** [reply]
I'm not that advanced yet! Welp, I guess I will have to wait till somebody makes a Luau thing.

**Comment by steve-chavez** [comment]
Looks like [Luau](https://luau-lang.org/) is a subset of Lua that includes a restricted set of libraries. So using a Lua library to connect directly to PostgreSQL it's not possible.

However, Luau has an HTTP client and also a json parser:
- https://developer.roblox.com/en-us/api-reference/class/HttpService
- https://developer.roblox.com/en-us/api-reference/function/HttpService/JSONDecode

Which should be enough for using Supabase!

If you're interested in creating a `postgrest-luau` library(which should be small), you can take inspiration from [postgrest-js](https://github.com/supabase/postgrest-js).

Also, you can ask for a Luau library on https://github.com/supabase/supabase/discussions/5, to see if anyone else is interested in collaborating.
